src/logion-chain/Vault.ts

```diff
@@ -0,0 +1,42 @@
+import { keyring } from "@polkadot/ui-keyring";
+import { RecoveryConfig } from "./Recovery";
+
+const THRESHOLD = 2;
+const NAME = "LogionVault";
+
+export function getVault(address: string, recoveryConfig: RecoveryConfig): string {
+    checkKeyringLoaded()
+    const vaultAddress = findVault();
+    return vaultAddress ? vaultAddress : createVault(address, recoveryConfig)
+}
+
+function createVault(address: string, recoveryConfig: RecoveryConfig): string {
+    const signatories: (string | Uint8Array)[] = [ address, ...recoveryConfig.friends.sort() ]
```

Comment:
In think that the whole array must be sorted, not only friends.

Reply:
Not sure we have the choice: https://github.com/paritytech/substrate/blob/v3.0.0/frame/multisig/src/lib.rs#L454